packages/clerk-js/src/utils/injectedWeb3Providers.ts

```diff
@@ -35,15 +35,23 @@ class InjectedWeb3Providers {
     coinbase: 'Coinbase Wallet',
     metamask: 'MetaMask',
   } as const;
+  static #instance: InjectedWeb3Providers | null = null;
 
-  constructor() {
+  private constructor() {
     if (typeof window === 'undefined') {
```

Comment:
Odd that this didn't catch it. Does this mean expo is defining `window`, but _not_ `addEventListener`? 😵

Reply:
React Native is defining `window` [here](https://github.com/facebook/react-native/blob/12b64b78247f2db929c6d421262bdec1f6402e0b/packages/react-native/Libraries/Core/setUpGlobals.js#L18) 🤯

Reply:
and looks like expo too

<img width="1009" alt="Screenshot 2024-09-03 at 3 10 26 PM" src="https://github.com/user-attachments/assets/f8e6e510-9b44-4d48-9cbb-669f6526e57b">